packages/netlify-cms-widget-markdown/src/MarkdownControl/plugins/List.js

```diff
@@ -268,9 +268,46 @@ function ListPlugin({ defaultType, unorderedListType, orderedListType }) {
 
         if (listOrListItem.type === 'list-item') {
           const listItem = listOrListItem;
-
+          console.log({ listItem });
+          const {
+            value: { document },
+          } = editor;
           // If focus is at start of list item, unwrap the entire list item.
           if (editor.atStartOf(listItem)) {
+            const parentList = document.getParent(listItem.key);
+            /* If the list item in question has a grandparent list, this means it is a child of a nested list.
+             * Htting Enter key on an empty nested list item like this should move that list item out of the nested list
+             * and into the gradparent list. The targeted list item becomes direct child of its grandparent list
+             * Example
+             * <ul> ----- CLOSEST ANCESTOR LIST
+             *  <li> ------ CLOESET ANCESTOR LIST ITEM
+             *    <p>foo</p>
+             *    <ul> ----- PARENT LIST
+             *      <li>
+             *        <p>bar</p>
+             *      </li>
+             *      <li>
+             *        <p></p> ----- WHERE THE ENTER KEY HAPPENS
+             *      </li>
+             *    </ul>
+             *  </li>
+             * </ul>
+             */
+            const closestAncestorList = document.getClosest(
+              parentList.key,
+              block => block.type === (LIST_TYPES[0] || LIST_TYPES[1]),
+            );
+            if (closestAncestorList) {
+              const closestAncestorListItem = document.getParent(parentList.key);
```

Comment:
Can you share a scenario to reach this code? I wasn't able to reach here

Reply:
First, let's comment out the whole `if (closestAncestorList)` block.
In the editor:
- Click Unorder List button, type “foo”. Hit Enter.
- Type “bar”. Hit Enter.
- Hit Tab to indent the list. Type “baz”. Hit Enter.

Expected:
```
<ul>
  <li>
    <p>foo</p></li>
  <li>
    <p>bar</p>
    <ul>
      <li>
        <p>baz</p>
      </li>
      <li>
        <p></p>
      </li>
    </ul>
  </li>
</ul>
```
Making sure that the cursor is inside the empty new list item below “baz”, hit Enter again.

This is when the condition `editor.atStartOf(listItem)` will be met. The original code unwraps the list item and turns it back into a paragraph block on the same level with the nested list that contains “baz”.

![new-paragraph-under-nested-list](https://user-images.githubusercontent.com/28924121/126589499-77312e85-e07d-40ac-8bb9-257a787b4810.png)

Now, hit Backspace to delete the paragraph and bring the cursor back to the end of “baz”.

Bring back the `closestAncestorList` we've commented out.

A list item always has a parent of type `bulleted-list` or `numbered-list`, but does this parent have another list block above it? The code will check that. If it is, we'll move the current list item outside to become a direct child of that list block from above.

But I can see why you're confused. My code will result in false negative for `closestAncestorList`. So let's fix it like this:

```
const closestAncestorList = document.getClosest(
  parentList.key,
  block => LIST_TYPES.includes(block.type),
);
```
Back to the editor. With the cursor at the end of “baz”, hit Enter twice. This is what should happen:
 
![move-nested-list-one-level-up](https://user-images.githubusercontent.com/28924121/126589240-a49bc1f1-49ab-45f9-9d16-7978fac74674.png)

Does that make sense?

Reply:
> Does that make sense?

Yes it does. The purpose is to create a new list item at the end of the parent list instead of a paragraph, when hitting enter of an empty sub list item for a non empty list.